qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -411,31 +489,39 @@ done
 testlog "TEST: disable mirror while daemon is stopped"
 stop_mirrors ${CLUSTER1}
 stop_mirrors ${CLUSTER2}
-set_pool_mirror_mode ${CLUSTER2} ${POOL} 'image'
+if [ "${RBD_MIRROR_MODE}" = "journal" ]; then
+  set_pool_mirror_mode ${CLUSTER2} ${POOL} 'image'
+fi
 disable_mirror ${CLUSTER2} ${POOL} ${image}
 if [ -z "${RBD_MIRROR_USE_RBD_MIRROR}" ]; then
   test_image_present ${CLUSTER1} ${POOL} ${image} 'present'
 fi
 start_mirrors ${CLUSTER1}
 wait_for_image_present ${CLUSTER1} ${POOL} ${image} 'deleted'
-set_pool_mirror_mode ${CLUSTER2} ${POOL} 'pool'
-enable_journaling ${CLUSTER2} ${POOL} ${image}
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  enable_mirror ${CLUSTER2} ${POOL} ${image}
+else
+  set_pool_mirror_mode ${CLUSTER2} ${POOL} 'pool'
+  enable_journaling ${CLUSTER2} ${POOL} ${image}
+fi
 wait_for_image_present ${CLUSTER1} ${POOL} ${image} 'present'
 wait_for_image_replay_started ${CLUSTER1} ${POOL} ${image}
 
 testlog "TEST: non-default namespace image mirroring"
 testlog " - replay"
-create_image ${CLUSTER2} ${POOL}/${NS1} ${image}
-create_image ${CLUSTER2} ${POOL}/${NS2} ${image}
-enable_mirror ${CLUSTER2} ${POOL}/${NS2} ${image} journal
+create_image_and_enable_mirror ${CLUSTER2} ${POOL}/${NS1} ${image}
+create_image_and_enable_mirror ${CLUSTER2} ${POOL}/${NS2} ${image}
+if [ "${RBD_MIRROR_MODE}" = "journal" ]; then
+  enable_mirror ${CLUSTER2} ${POOL}/${NS2} ${image} journal
```

Comment:
Yes, True

Reply:
How did you resolve this?  I guess you identified that `create_image` -> `create_image_and_enable_mirror`  did not make ` enable_mirror ${CLUSTER2} ${POOL}/${NS2} ${image} journal` redundant after all. `enable_mirror` gets called within `create_image_and_enable_mirror` only `if [ "${MIRROR_POOL_MODE}" = "image" ] || [ "$pool" = "${PARENT_POOL}" ]"`  and when mirroring mode is journal we set $MIRROR_POOL_MODE as pool. Is my understanding correct?

Reply:
correct

Reply:
I think it is rather a bug in `create_image_and_enable_mirror` that should be fixed. It should check for namespace and if it is `NS2` then it should enable mirror, because `NS2` is setup to "image" mode (similarly to the parent pool).

Reply:
Sure, sounds reasonable.